dids/diddht/internal/dns/did.go

```diff
@@ -0,0 +1,299 @@
+package dns
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+	"github.com/tbd54566975/web5-go/dids/didcore"
+	"golang.org/x/net/dns/dnsmessage"
+)
+
+// MarshalDIDDocument packs a DID document into a TXT DNS resource records and adds to the DNS message Answers
+func MarshalDIDDocument(d *didcore.Document) ([]byte, error) {
+
+	// create root record
+	var msg dnsmessage.Message
+
+	// get sorted VM IDs
+	sortedIDs := pluckSort(d.VerificationMethod)
+	vmIDToK := map[string]string{}
+
+	vmBep44Keys := []string{}
+
+	for k, id := range sortedIDs {
+		_k := fmt.Sprintf("k%d", k)
+		vmIDToK[id] = _k
+		vmBep44Keys = append(vmBep44Keys, _k)
+	}
+
+	sToK := map[string]string{}
+	sKeys := []string{}
+	for k, v := range d.Service {
+		_k := fmt.Sprintf("s%d", k)
+		sToK[v.ID] = _k
+		sKeys = append(sKeys, _k)
+	}
+
+	rootProps := map[string][]string{
+		"v":    {"1"},
+		"id":   {d.ID},
+		"vm":   vmBep44Keys,
+		"auth": methodsToKeys(d.Authentication, vmIDToK),
+		"asm":  methodsToKeys(d.AssertionMethod, vmIDToK),
+		"agm":  methodsToKeys(d.KeyAgreement, vmIDToK),
+		"inv":  methodsToKeys(d.CapabilityInvocation, vmIDToK),
+		"del":  methodsToKeys(d.CapabilityDelegation, vmIDToK),
+		"srv":  sKeys,
+		"cnt":  d.Controller,
+		"aka":  d.AlsoKnownAs,
```

Comment:
should move these field names to consts if they're used elsewhere. for example i see `asm` used in 3 places